ksqldb-functional-tests/src/test/resources/historical_plans/array_-_GENERATE_SERIES/5.5.0_1581572083765/spec.json

```diff
@@ -73,6 +73,23 @@
     } ],
     "statements" : [ "CREATE STREAM TEST (F0 INT, F1 INT) WITH (kafka_topic='test_topic', value_format='JSON');", "CREATE STREAM OUTPUT AS SELECT GENERATE_SERIES(F0, F1) FROM TEST;" ],
     "post" : {
+      "sources" : [ {
+        "name" : "OUTPUT",
+        "type" : "STREAM",
+        "schema" : "`ROWKEY` STRING KEY, `KSQL_COL_0` ARRAY<INTEGER>",
+        "keyFormat" : {
+          "format" : "KAFKA"
+        },
+        "serdeOptions" : [ ]
```

Comment:
I wouldn't expect this `serdeOptions` to be included if its empty due to the `@JsonInclude(JsonInclude.Include.NON_EMPTY` annotation.

Did you maybe generate there _before_ adding annotation?

Might be worth removing these to avoid changes if someone regenerates later. Plus it will also test the deserialization works when this field is not present.

Reply:
> I wouldn't expect this serdeOptions to be included if its empty due to the @JsonInclude(JsonInclude.Include.NON_EMPTY annotation.

That's funny, I thought I added `NON_NULL` but it appears I added `NON_EMPTY`. From the new unit tests in SourceNodeTest.java, both function the same way for the optional set field, maybe because non-empty = non-null for optionals? I'll update it to NON_NULL to avoid confusion.

> Did you maybe generate there before adding annotation?

Nope, generated after. I wanted the empty serde options in the historic plan specs since lack of serde options means no validation on serde options is performed, whereas empty serde options means the test validates that the set of serde options is empty.

Reply:
I think NON_EMPTY probably makes more sense, as this wouldn't output `"serdeOptions" : [ ]` where the optional was empty.

Reply:
NON_EMPTY and NON_NULL behave the same way for the `Optional<Set<SerdeOption>>` type: both output `"serdeOptions" : [ ]` where the optional contains an empty set, and do not output `serdeOptions` when the optional is empty. See the new unit tests in `SourceNodeTest` for confirmation.

Reply:
Ah but there is a slight difference: NON_NULL outputs `"serdeOptions": null` if the Optional is empty, whereas NON_EMPTY leaves out serdeOptions entirely (which is maybe what you meant to say). That's definitely good reason for switching back to NON_EMPTY -- will do.